src/CMake/findpackage.cmake

```diff
@@ -0,0 +1,60 @@
+message("-- Preparing XRT find_package")
+
+include(CMakePackageConfigHelpers)
+
+string(TOLOWER ${PROJECT_NAME} LOWER_NAME)
+
+file(RELATIVE_PATH REL_INCLUDE_DIR
+  ${XRT_INSTALL_DIR}/share/cmake/${PROJECT_NAME}
+  ${XRT_INSTALL_DIR}/include
+  )
+
+set(CONF_INCLUDE_DIRS "\${${PROJECT_NAME}_CMAKE_DIR}/${REL_INCLUDE_DIR}")
+
+set(CONF_EXPORT_GROUPS xrt_core xilinxopencl xrt_swemu xrt_hwemu)
+
+set(CONF_TARGETS xrt_core xrt_coreutil xilinxopencl xrt++ xrt_swemu xrt_hwemu)
+
+configure_file (
+  ${CMAKE_SOURCE_DIR}/CMake/config/xrt.fp.in
+  ${CMAKE_CURRENT_BINARY_DIR}/${LOWER_NAME}-config.cmake
+  @ONLY
+  )
+write_basic_package_version_file (
+  ${CMAKE_CURRENT_BINARY_DIR}/${LOWER_NAME}-config-version.cmake
+  VERSION ${XRT_VERSION_STRING}
+  COMPATIBILITY AnyNewerVersion
+  )
+install (
+  FILES ${CMAKE_CURRENT_BINARY_DIR}/${LOWER_NAME}-config.cmake ${CMAKE_CURRENT_BINARY_DIR}/${LOWER_NAME}-config-version.cmake
+  DESTINATION ${XRT_INSTALL_DIR}/share/cmake/${PROJECT_NAME}
+  COMPONENT ${XRT_DEV_COMPONENT}
+  )
+
+if(TARGET xrt_core)
```

Comment:
This doesn't look like it scales very well.  How will we remember to update this file when adding new targets?

Reply:
as far as i know there is no way to automatically discover available targets in a cmake project. I believe this has to be added every time you add a new library. Perhaps it could be put in the library's CMakeLists.txt its self?

I'll double check on my above assertion.